pkg/pillar/vendor/github.com/digitalocean/go-qemu/qmp/socket.go

```diff
@@ -135,13 +135,29 @@ func (mon *SocketMonitor) Connect() error {
 		return err
 	}
 
-	// Check for no error on return
-	var r response
-	if err := dec.Decode(&r); err != nil {
-		return err
-	}
-	if err := r.Err(); err != nil {
-		return err
+	// Wait for response
+	scanner := bufio.NewScanner(mon.c)
+	for scanner.Scan() {
```

Comment:
Could there be a risk of an infinite loop if non-capability events keep coming in without any actual progress?

Reply:
The whole implementation of this library does not imply the timeout on response. So we hang here and on the Run() (commands invocation path) as well if QEMU qmp loop hangs.